src/IO/ReadBufferFromPocoSocket.cpp

```diff
@@ -27,23 +28,23 @@ bool ReadBufferFromPocoSocket::nextImpl()
     ssize_t bytes_read = 0;
     Stopwatch watch;
 
-    int flags = 0;
-    if (async_callback)
-        flags |= MSG_DONTWAIT;
```

Comment:
BTW @tavplubix added this to `SecureSocket`: `if ((flags & MSG_DONTWAIT) && (flags & MSG_PEEK))` so maybe just adding extra flag may help.

Reply:
`MSG_PEEK: Peeks at an incoming message. The data is treated as unread and the next recv() or similar function shall still return this data.`
I guess this flag is inappropriate here, because we don't want to treated data as unread after reading.